PWGHF/D2H/Tasks/taskDstarToD0Pi.cxx

```diff
@@ -15,18 +15,86 @@
+#include "Framework/Expressions.h"
+#include "Framework/HistogramRegistry.h"
 #include "Framework/runDataProcessing.h"
 
+// O2Physics
+#include "Common/DataModel/Centrality.h"
+
+// PWGHF
 #include "PWGHF/Core/SelectorCuts.h"
 #include "PWGHF/DataModel/CandidateReconstructionTables.h"
 #include "PWGHF/DataModel/CandidateSelectionTables.h"
 
+namespace o2::aod
+{
+namespace hf_cent_mc_dstar
+{
+DECLARE_SOA_COLUMN(CentMcCollision, centMcCollision, float);
+}
+DECLARE_SOA_TABLE(HfCentMcCollisions, "AOD", "HfCentMcCol", hf_cent_mc_dstar::CentMcCollision); // Joinable with McCollisions
+using HfCentMcCollision = HfCentMcCollisions::iterator;
+} // namespace o2::aod
+
 using namespace o2;
 using namespace o2::analysis;
 using namespace o2::framework;
 using namespace o2::framework::expressions;
+using namespace o2::soa;
+
+struct HfCentMcDstarToD0Pi {
+  Produces<aod::HfCentMcCollisions> rowsCentMcCollision;
+
+  using CollisionsWCentMcLabel = o2::soa::Join<o2::aod::Collisions, o2::aod::McCollisionLabels, o2::aod::CentFT0Ms>;
+
+  SliceCache cache;
+
+  // Unsorted Slicing is working and also faster than partition defined within process function
+  PresliceUnsorted<CollisionsWCentMcLabel> recColsPerMcColPreSlice = aod::mccollisionlabel::mcCollisionId;
+
+  HistogramRegistry registry{
+    "registry",
+    {{"QA/hCentGen", "#{Delta}Cent % distribution of Collisions having same MC Collision;FT0M Centrality %; Counts", {HistType::kTH1F, {{100, 0.0, 100.0}}}}}};
+
+  void process(aod::McCollisions const& mcCollisions, CollisionsWCentMcLabel const& collisions)
+  {
+    for (const auto& mcCollision : mcCollisions) {
+      auto gIMcCollision = mcCollision.globalIndex();
+      auto recColsPerMcCol = collisions.sliceByCachedUnsorted(aod::mccollisionlabel::mcCollisionId, gIMcCollision, cache);
+
+      // LOGF(info,"recColsPerMcCol size: %d",recColsPerMcCol.size());
+      // looking if a generated collision reconstructed more than a times.
+      if (recColsPerMcCol.size() > 1) {
+        for (const auto& [c1, c2] : combinations(CombinationsStrictlyUpperIndexPolicy(recColsPerMcCol, recColsPerMcCol))) {
+          auto deltaCent = abs(c1.centFT0M() - c2.centFT0M());
+          registry.fill(HIST("QA/hCentGen"), deltaCent);
+        }
+      }
+
+      if (recColsPerMcCol.size()) {
+        // assigning centrality to MC Collision using max FT0M amplitute from Reconstructed collisions
+        std::vector<float> colCentVec;
+        for (const auto& recCol : recColsPerMcCol) {
+          colCentVec.push_back(recCol.centFT0M());
+        }
+        std::sort(colCentVec.begin(), colCentVec.end());
+        auto centFT0MGen = colCentVec[colCentVec.size() - 1];
+
+        // filling centrality table at gen level if collision was reconstructed at REC level successfully
+        rowsCentMcCollision(centFT0MGen);
+      } else {
+        // filling centrality table at gen level if collision was not reconstructed at REC level
+        rowsCentMcCollision(0.0);
```

Comment:
put a negative value if the collision is not reconstructed, since `0` is still a value in the valid range

Reply:
Hi @fgrosa, I created a [new PR](https://github.com/AliceO2Group/O2Physics/pull/6957) for all the above changes suggested. Please have a look at it. Thanks.